### PR TITLE
OCPBUGS-38557: Add support for GCP Workload Identity / Federated identity operator installs

### PR DESCRIPTION
**Pull request by cyril-ui-developer**
### After:

Screenshot 2024-07-26 at 10 57 14 AM

Screenshot 2024-07-30 at 10 18 37 AM

Screenshot 2024-07-30 at 10 19 46 AM

Screenshot 2024-07-26 at 11 00 50 AM

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 1]
```diff
@@ -814,6 +881,15 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
               )}
             </p>
           </Alert>
+        )}{' '}
+        {tokenizedAuth === 'GCP' && showSTSWarn && (
+          <STSWarningAlert
```
Comment (jhadvig) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:886]: I know it not new, but lets swap the Alerts above with the new `STSWarningAlert`.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 2]
```diff
@@ -904,6 +980,50 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
                   </fieldset>
                 </div>
               )}
+              {tokenizedAuth === 'GCP' && (
+                <div className="form-group">
+                  <InputField
+                    label={t('olm~GCP Project Number')}
+                    helpText={t(
+                      'olm~The GCP PRoject Number required for the operator to access the cloud API.',
```
Comment (jhadvig) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:988]:
```suggestion
                      'olm~The GCP Project Number required for the operator to access the cloud API.',
```

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx [hunk 3]
```diff
@@ -413,6 +413,24 @@ export const OperatorHubItemDetails: React.FC<OperatorHubItemDetailsProps> = ({
                     </p>
                   </Alert>
                 )}
+              {/* TODO: Remove the negation signs for !isGCPWIFCluster && !infraFeatures */}
+              {!isGCPWIFCluster(cloudCredentials, infrastructure, authentication) &&
+                showWarn &&
+                !infraFeatures?.find((i) => i === InfraFeatures.tokenAuthGCP) && (
+                  <Alert
```
Comment (jhadvig) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx:420]: lets use `STSWarningAlert`

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 4]
```diff
@@ -814,6 +881,15 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
               )}
             </p>
           </Alert>
+        )}{' '}
+        {tokenizedAuth === 'GCP' && showSTSWarn && (
+          <STSWarningAlert
+            title={t('olm~Cluster in GCP Workload Identity / Federated Identity Mode')}
+            message={t(
+              'olm~This cluster is using GCP Workload Identity / Federated Identity to reach the cloud API. In order for this operator to take the actions it requires directly with the cloud API, provide the Client ID, Tenant ID, and Subscription ID during installation. Manual subscriptions are highly recommended as steps should be taken before upgrade to ensure that the permissions required by the next version are properly accounted for in the role. See the operator description for more details.',
```
Comment (jhadvig) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:889]: The GCP WIF will have different variable names:

Project ID
Pool ID
Provider ID
Service Account Email

**Review by jhadvig — COMMENTED** [review]
Very nice and fast progress on the work @cyril-ui-developer 👍 
Hope you dont mind adding a review even if the PR is WIP. Just couple of nits.

File: frontend/packages/operator-lifecycle-manager/locales/en/olm.json [hunk 5]
```diff
@@ -332,6 +335,12 @@
   "The Azure Tenant ID required for the operator to access the cloud API.": "The Azure Tenant ID required for the operator to access the cloud API.",
   "Azure Subscription ID": "Azure Subscription ID",
   "The Azure Subscription ID required for the operator to access the cloud API.": "The Azure Subscription ID required for the operator to access the cloud API.",
+  "GCP Tenant ID": "GCP Tenant ID",
```
Comment (jhadvig) [inline frontend/packages/operator-lifecycle-manager/locales/en/olm.json:338]: Tenant ID is only needed for Azure WIF. We dont need it for GCP.

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx [hunk 6]
```diff
@@ -785,123 +856,117 @@ export const OperatorHubSubscribeForm: React.FC<OperatorHubSubscribeFormProps> =
...
+                    ariaLabel={'Azure Subcription ID'}
+                    value={azureSubscriptionId}
+                    setValue={setAzureSubscriptionId}
+                  />
+                </div>
+              )}
+              {tokenizedAuth === 'GCP' && (
+                <div className="form-group">
+                  <InputField
+                    label={t('olm~GCP Tenant ID')}
```
Comment (jhadvig) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-subscribe.tsx:941]:
```suggestion
                    label={t('olm~GCP Pool ID')}
```

**Review by jhadvig — CHANGES_REQUESTED** [review]
Thanks @cyril-ui-developer the PR looks good 👍 
Adding two comments, otherwise I think its good to go :)

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Comment by XiyunZhao** [comment]
Working on this Pr, still need sometime to check

**Comment by XiyunZhao** [comment]
@cyril-ui-developer could you help to double-check if following issues need a bug to track?
1. The Warning message is missing on Operator installation page 
2. The update approval strategy should be set to 'Manual' by default
Here are the details:
Console4176_warningmissing

**Comment by cyril-ui-developer** [comment]
> @cyril-ui-developer could you help to double-check if following issues need a bug to track?
> 
> 1. The Warning message is missing on Operator installation page
> 2. The update approval strategy should be set to 'Manual' by default
>    Here are the details:
>    Console4176_warningmissing
>          
>            [
>              
>            
>                Console4176_warningmissing
>              ](https://private-user-images.githubusercontent.com/78589509/355274904-a1836173-0c33-421c-9c54-9ae884f2b1f8.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5MDg0MTIsIm5iZiI6MTcyMjkwODExMiwicGF0aCI6Ii83ODU4OTUwOS8zNTUyNzQ5MDQtYTE4MzYxNzMtMGMzMy00MjFjLTljNTQtOWFlODg0ZjJiMWY4LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA2VDAxMzUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3NGYzZmQwOTgxNWEwNmQwOWY1NjdlYWZjMDE2YzQzYzlkYjVlNTVkNWFjODNkZGYxZjIwMzIzZTQyNTEzYTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.zVI_z4ZY8jxeQUqPqdduVEtKr8M5l2pYcNlo756A7X0)
>            
>            
>              
>                
>                  
>                
>                
>                  
>                  
>                
>              
>              [
>                
>                  
>                
>              ](https://private-user-images.githubusercontent.com/78589509/355274904-a1836173-0c33-421c-9c54-9ae884f2b1f8.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5MDg0MTIsIm5iZiI6MTcyMjkwODExMiwicGF0aCI6Ii83ODU4OTUwOS8zNTUyNzQ5MDQtYTE4MzYxNzMtMGMzMy00MjFjLTljNTQtOWFlODg0ZjJiMWY4LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA2VDAxMzUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3NGYzZmQwOTgxNWEwNmQwOWY1NjdlYWZjMDE2YzQzYzlkYjVlNTVkNWFjODNkZGYxZjIwMzIzZTQyNTEzYTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.zVI_z4ZY8jxeQUqPqdduVEtKr8M5l2pYcNlo756A7X0)

Actually, I tested this with mock data because we couldn't reproduce the environment locally. Could I get access to the cluster you used for verification?

**Comment by XiyunZhao** [comment]
@cyril-ui-developer Let me know when you need the cluster, I will prepare one for you

**Comment by cyril-ui-developer** [comment]
> @cyril-ui-developer Let me know when you need the cluster, I will prepare one for you

8 a.m. (ET). Thank you.

> @cyril-ui-developer Let me know when you need the cluster, I will prepare one for you



> @cyril-ui-developer Let me know when you need the cluster, I will prepare one for you

8

**Comment by cyril-ui-developer** [comment]
> @cyril-ui-developer could you help to double-check if following issues need a bug to track?
> 
> 1. The Warning message is missing on Operator installation page
> 2. The update approval strategy should be set to 'Manual' by default
>    Here are the details:
>    Console4176_warningmissing
>          
>            [
>              
>            
>                Console4176_warningmissing
>              ](https://private-user-images.githubusercontent.com/78589509/355274904-a1836173-0c33-421c-9c54-9ae884f2b1f8.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5NDA4MDcsIm5iZiI6MTcyMjk0MDUwNywicGF0aCI6Ii83ODU4OTUwOS8zNTUyNzQ5MDQtYTE4MzYxNzMtMGMzMy00MjFjLTljNTQtOWFlODg0ZjJiMWY4LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA2VDEwMzUwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwMjU3MmY5ZDkzZWFmNTU0NjUyYzc5YzJkNTI3ZWE3NDBhNjdiMDBkYWMyNTc1NjgzY2JmNDc1NDBjNjZkMmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.01x0zFUiwtbK-QYcfevwxljamkpueLzn-EUjmWPg0cY)
>            
>            
>              
>                
>                  
>                
>                
>                  
>                  
>                
>              
>              [
>                
>                  
>                
>              ](https://private-user-images.githubusercontent.com/78589509/355274904-a1836173-0c33-421c-9c54-9ae884f2b1f8.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5NDA4MDcsIm5iZiI6MTcyMjk0MDUwNywicGF0aCI6Ii83ODU4OTUwOS8zNTUyNzQ5MDQtYTE4MzYxNzMtMGMzMy00MjFjLTljNTQtOWFlODg0ZjJiMWY4LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA2VDEwMzUwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwMjU3MmY5ZDkzZWFmNTU0NjUyYzc5YzJkNTI3ZWE3NDBhNjdiMDBkYWMyNTc1NjgzY2JmNDc1NDBjNjZkMmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.01x0zFUiwtbK-QYcfevwxljamkpueLzn-EUjmWPg0cY)

@XiyunZhao This is happening because the value of `tokenizedAuth` is `null` for the cert-manager operator. The conditions for displaying the warning on the modal are slightly different from those on the subscription page. As mentioned earlier, we were unable to verify this with actual data. Are you able to verify the same behavior for Azure or AWS using your environment? That might help us figure out what is missing. cc: @jhadvig

**Comment by cyril-ui-developer** [comment]
Docs Approver:
/assign @opayne1

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx [hunk 7]
```diff
@@ -380,38 +379,37 @@ export const OperatorHubItemDetails: React.FC<OperatorHubItemDetailsProps> = ({
...
-                    className="pf-v5-u-mb-lg"
-                  >
-                    <p>
-                      {t(
-                        'olm~This cluster is using AWS Security Token Service to reach the cloud API. In order for this operator to take the actions it requires directly with the cloud API, you will need to provide a role ARN (with an attached policy) during installation. Please see the operator description for more details.',
-                      )}
-                    </p>
-                  </Alert>
+                    message={t(
+                      'olm~This cluster is using AWS Security Token Service to reach the cloud API. In order for this operator to take the actions it requires directly with the cloud API, you will need to provide a role ARN (with an attached policy) during installation. Please see the operator description for more details.',
```
Comment (opayne1) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-item-details.tsx:387]: Small wording suggestion
```suggestion
                      'olm~This cluster is using AWS Security Token Service to reach the cloud API. In order for this operator to take the actions it requires directly with the cloud API, you must provide a role ARN (with an attached policy) during installation. Please see the operator description for more details.',
```

**Review by opayne1 — COMMENTED** [review]
One small suggestion but otherwise lgtm!

/label docs-approved

**Comment by XiyunZhao** [comment]
@cyril-ui-developer I checked on the AWS cluster, it's working as normal, so could you help to check the issue on this PR. also just a notice that we are nearing the branching date
Here is the correct behavior for STS mode :
CONSOLE4176_aws

**Comment by cyril-ui-developer** [comment]
> @cyril-ui-developer could you help to double-check if following issues need a bug to track?
> 
> 1. The Warning message is missing on Operator installation page
> 2. The update approval strategy should be set to 'Manual' by default
>    Here are the details:
>    Console4176_warningmissing
>          
>            [
>              
>            
>                Console4176_warningmissing
>              ](https://private-user-images.githubusercontent.com/78589509/355274904-a1836173-0c33-421c-9c54-9ae884f2b1f8.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwNDUxMjUsIm5iZiI6MTcyMzA0NDgyNSwicGF0aCI6Ii83ODU4OTUwOS8zNTUyNzQ5MDQtYTE4MzYxNzMtMGMzMy00MjFjLTljNTQtOWFlODg0ZjJiMWY4LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA3VDE1MzM0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3NjE4ZjYxYjkwMjcyNmQwYWE3MzI3MGZhNGE1N2U2OGFmMmFmMWIyZDU2MDllMTQ4ZmRlNDAwNTVmNDVmNmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.q1hRKMtj23tF37Lop5nUclchZ9uhgNIzHAyF4pF7DKw)
>            
>            
>              
>                
>                  
>                
>                
>                  
>                  
>                
>              
>              [
>                
>                  
>                
>              ](https://private-user-images.githubusercontent.com/78589509/355274904-a1836173-0c33-421c-9c54-9ae884f2b1f8.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwNDUxMjUsIm5iZiI6MTcyMzA0NDgyNSwicGF0aCI6Ii83ODU4OTUwOS8zNTUyNzQ5MDQtYTE4MzYxNzMtMGMzMy00MjFjLTljNTQtOWFlODg0ZjJiMWY4LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA3VDE1MzM0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3NjE4ZjYxYjkwMjcyNmQwYWE3MzI3MGZhNGE1N2U2OGFmMmFmMWIyZDU2MDllMTQ4ZmRlNDAwNTVmNDVmNmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.q1hRKMtj23tF37Lop5nUclchZ9uhgNIzHAyF4pF7DKw)

@XiyunZhao Fixed and updated. PTAL.
Good catch :). Thank you.

**Comment by cyril-ui-developer** [comment]
cc: @jhadvig PTAL and tag again.

**Comment by XiyunZhao** [comment]
@cyril-ui-developer Thanks for the quick fix, I need a confirmation about the customized field on the installation page.
The field 'PROJECT_ID' is missing on the installation page, which does not satisfy the description of our design. Is there any design change on this part? Kindly help to take a look~
image

**Comment by jhadvig** [comment]
@XiyunZhao yes the PROJECT_ID is missing on purpose, there was a change in the design, which was communicated by @dlom, that only Pool_ID, Provider_ID and SAEmail are required.

**Comment by XiyunZhao** [comment]
@jhadvig Thanks for the quick reply, no other issue was found on this PR
/label qe-approved

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by cyril-ui-developer** [comment]
@XiyunZhao I rebased this PR due to recent changes upstream. I verified with mock data, but ideally it would be good to retest the changes with real data. If possible, could you please retest?

**Comment by dlom** [comment]
@jhadvig the document has been updated again: we have combined all fields (except email) into a single AUDIENCE field, which is made up of the 3 original parts (pool, provider, and project)

**Comment by XiyunZhao** [comment]
@cyril-ui-developer @jhadvig Kindly help to check the failure, and hope this PR can be merged

**Pull request merged**